modules/apps/portal-vulcan/portal-vulcan-impl/src/test/java/com/liferay/portal/vulcan/internal/exportimport/data/handler/VulcanBatchEnginePortletDataHandlerTest.java

```diff
@@ -0,0 +1,163 @@
+
+	@ClassRule
+	@Rule
+	public static final LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@Before
+	public void setUp() throws Exception {
+		_vulcanBatchEnginePortletDataHandler =
+			new VulcanBatchEnginePortletDataHandler(
+				_batchEngineExportTaskExecutor, _batchEngineExportTaskService,
+				null, null, RandomTestUtil.randomString(),
+				RandomTestUtil.randomString());
+	}
+
+	@Test
+	public void testBuildFilterParameterWithNoDates() {
+		Date endDate = null;
+		Date startDate = null;
+
+		PortletDataContext portletDataContext = _mockPortletDataContext(
+			endDate, startDate);
+
+		Map<String, Serializable> parameterMap =
+			_vulcanBatchEnginePortletDataHandler.buildParameterMap(
+				portletDataContext);
+
+		Assert.assertEquals(StringPool.BLANK, parameterMap.get("filter"));
+	}
+
+	@Test
+	public void testBuildFilterParameterWithOnlyEndDate() throws Exception {
+		Date endDate = _getDate(0);
+		Date startDate = null;
+
+		PortletDataContext portletDataContext = _mockPortletDataContext(
+			endDate, startDate);
+
+		Map<String, Serializable> parameterMap =
+			_vulcanBatchEnginePortletDataHandler.buildParameterMap(
+				portletDataContext);
+
+		Assert.assertEquals(
+			"dateModified le " + _dateFormat.format(endDate),
+			parameterMap.get("filter"));
+	}
+
+	@Test
+	public void testBuildFilterParameterWithOnlyStartDate() throws Exception {
+		Date endDate = null;
+		Date startDate = _getDate(-1);
+
+		PortletDataContext portletDataContext = _mockPortletDataContext(
+			endDate, startDate);
+
+		Map<String, Serializable> parameterMap =
+			_vulcanBatchEnginePortletDataHandler.buildParameterMap(
+				portletDataContext);
+
+		Assert.assertEquals(
+			"dateModified ge " + _dateFormat.format(startDate),
+			parameterMap.get("filter"));
+	}
+
+	@Test
+	public void testBuildFilterParameterWithStartAndEndDate() throws Exception {
+		Date endDate = _getDate(0);
+		Date startDate = _getDate(-1);
+
+		PortletDataContext portletDataContext = _mockPortletDataContext(
+			endDate, startDate);
+
+		Map<String, Serializable> parameterMap =
+			_vulcanBatchEnginePortletDataHandler.buildParameterMap(
+				portletDataContext);
+
+		Assert.assertEquals(
+			StringBundler.concat(
+				"dateModified ge ", _dateFormat.format(startDate),
+				" and dateModified le ", _dateFormat.format(endDate)),
+			parameterMap.get("filter"));
+	}
+
+	private Date _getDate(int addDays) {
+		Calendar calendar = Calendar.getInstance();
+
+		calendar.add(Calendar.DATE, addDays);
+
+		return calendar.getTime();
+	}
+
+	private PortletDataContext _mockPortletDataContext(
+		Date endDate, Date startDate) {
+
+		PortletDataContext portletDataContext = Mockito.mock(
+			PortletDataContext.class);
+
+		Mockito.when(
+			portletDataContext.getStartDate()
+		).thenReturn(
+			startDate
+		);
+
+		Mockito.when(
+			portletDataContext.getEndDate()
+		).thenReturn(
+			endDate
+		);
+
+		return portletDataContext;
+	}
+
+	@Mock
```

Comment:
I think this annotation is not necessary if you don't call the method `MockitoAnnotations.initMocks(this);`. You are initializing it to a mock already